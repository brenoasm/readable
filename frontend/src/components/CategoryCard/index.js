import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const propTypes = {
  category: PropTypes.shape({
    name: PropTypes.string.isRequired,
    path: PropTypes.string.isRequired
  }).isRequired
};

const StyledCategoryCard = styled.div`
  margin: 10px auto;
  border: #000 solid 1px;
`;

const CategoryCard = ({ category }) => (
  <StyledCategoryCard>
    <h2>{category.name}</h2>
    <Link to={category.path}>Ver os detalhes</Link>
  </StyledCategoryCard>
)

CategoryCard.propTypes = propTypes;

export default CategoryCard;
