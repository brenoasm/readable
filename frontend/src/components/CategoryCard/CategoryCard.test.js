/* eslint-disable */

import React from 'react';

import CategoryCard from './';

describe('CategoryCard', () => {
  let wrapper;

  beforeEach(() => {
    const category = {
      name: 'redux',
      path: 'redux'
    };

    wrapper = shallow(<CategoryCard category={category} />);
  })

  afterEach(() => {
    wrapper.unmount();
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});
