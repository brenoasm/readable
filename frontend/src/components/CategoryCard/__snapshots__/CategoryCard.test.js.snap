// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategoryCard
    category={
        Object {
            "name": "redux",
            "path": "redux",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h2>
          redux
</h2>,
        <Link
          replace={false}
          to="redux"
>
          Ver os detalhes
</Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "redux",
        },
        "ref": null,
        "rendered": "redux",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Ver os detalhes",
          "replace": false,
          "to": "redux",
        },
        "ref": null,
        "rendered": "Ver os detalhes",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h2>
            redux
</h2>,
          <Link
            replace={false}
            to="redux"
>
            Ver os detalhes
</Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "redux",
          },
          "ref": null,
          "rendered": "redux",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Ver os detalhes",
            "replace": false,
            "to": "redux",
          },
          "ref": null,
          "rendered": "Ver os detalhes",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
